packages/flutter_tools/lib/src/commands/create_base.dart

```diff
@@ -156,37 +168,8 @@ abstract class CreateBase extends FlutterCommand {
   }
 
   /// Gets the flutter root directory.
-  ///
-  /// Throw with exit code 2 if the flutter sdk installed is invalid.
   @protected
-  String get flutterRoot {
```

Comment:
This is completely unnecessary and I couldn't figure out what it is doing :)

Reply:
A wild guess, but Is it needed for --local-engine? Are there checks elsewhere that `throwToolExit` when `Cache.flutterRoot` is null? It looks like these are very conservative sanity checks, were they added in response to a github issue?

Reply:
localEngine doesn't modify FlutterRoot though 🤔

Reply:
I traced it back as far as https://github.com/flutter/flutter/commit/cccd917ac25b043b84449096ccbe51ab0492a85c

My guess is we used to be less careful about setting the flutter root to something valid, but the logic is much more solid (and tested !) now